### PR TITLE
WIP: Sync file attributes on Windows OS. (fixes #4935)

### LINKED ISSUE #4935: Enforcing hidden file attribute in the Windows world
I have to sync several Windows machines where some directory have many files that have the HIDDEN attribute bit set.

On the destination machines, after the synchronization is done, the attribute is lost, so I must resort to use some scripts to keep the files synced with this attribute.
I see that the HIDDEN attribute is already used in the Windows version of Synchthing, for hiding the temporary files under sync, but then it is removed on final rename.

Having this attribute enforced in this platform will be very beneficial, at least for who use this very useful tool in this os platform that is very popular.

Using v0.14.47, Windows (64 bit)

### PR DESCRIPTION
**Pull request by Kropekk**
### Purpose

Add Attribute field to the FileInfo protocol and handle it properly on Windows OS. Currently only 3 attributes will be handled: FILE_ATTRIBUTE_HIDDEN, FILE_ATTRIBUTE_SYSTEM, FILE_ATTRIBUTE_NOT_CONTENT_INDEXED. See: https://forum.syncthing.net/t/enforcing-hidden-file-attribute-in-the-windows-world/11574/

### Testing

Currently no tests, however it is easy to imagine some basic ones.

### Documentation
If pulled, the protocol will change, so it will be necessary to update documentation, for example: https://docs.syncthing.net/specs/bep-v1.html#id1

### CODE REVIEW

File: lib/fs/basicfs_windows.go [hunk 1]
```diff
@@ -122,6 +123,56 @@ func (f *BasicFilesystem) Hide(name string) error {
...
 }
 
+// Typically, attrs should be populated with value from GetFileAttributes and some bitwise operations performed on them
+func (f *BasicFilesystem) SetFileAttributes(name string, attrs uint32) error {
+	name, err := f.rooted(name)
+	if err != nil {
+		return err
+	}
+
+	p, err := syscall.UTF16PtrFromString(name)
```
[the remark this answers is not in the document]
Comment (Kropekk) [reply lib/fs/basicfs_windows.go]: I just copied this part from Hide/Unhide, but of course I can make a change to use x/sys.

**Comment by calmh** [comment]
We need to specify what the bits in the attribute field means, so those values will become part of the protocol. That probably won't be the same bit values as used by the syscalls etc so there will need to be translation in the fs layer.

**Comment by Kropekk** [comment]
>  That probably won't be the same bit values as used by the syscalls etc

Could you explain why?

**Comment by st-review** [comment]
:robot: *beep boop*

I'm going to close this pull request as it has been idle for more than 90 days.

This is not a rejection, merely a removal from the list of active pull requests that is periodically reviewed by humans. The pull request can be reopened when there is new activity, and merged once any remaining issues are resolved.

**Comment by p0larstern** [comment]
Is this pull request and it's related issue are still WIP, also does this effect the aim and requirement of issue #5162 ?

**Pull request closed without merging**